### PR TITLE
Refactor for downstream usage

### PR DESCRIPTION
**Pull request by floryst**
This PR addresses #126, #107, and #110. Glance currently does not have a developer-friendly way of using the top-level App component when embedding inside another vue application.

### CODE REVIEW

**Comment by floryst** [comment]
@zachmullen can you take a brief look to make sure this adequately addresses downstream usage?

**Comment by floryst** [comment]
@zachmullen can you take a brief look to make sure this adequately addresses downstream usage?

File: src/app.js [hunk 1]
```diff
@@ -51,31 +44,18 @@ export function createViewer(container, proxyConfig = null) {
...
-      proxyManager,
-      widgetManager,
+    store: Store,
+    data() {
+      return {
+        proxyManager,
+      };
     },
-    template: '<App ref="app" />',
+    template: '<App ref="app" :proxyManager="proxyManager" />',
```
Comment (zachmullen) [inline src/app.js]: This `ref` can go away now.

File: src/components/core/App/script.js [hunk 2]
```diff
@@ -31,9 +41,43 @@ export default {
...
+    },
+    proxyManager: {
+      required: false,
+    },
+  },
   data() {
+    let proxyManager = this.proxyManager;
+    if (!proxyManager) {
+      proxyManager = vtkProxyManager.newInstance({ proxyConfiguration });
+    }
```
Comment (zachmullen) [inline src/components/core/App/script.js]: Vue allows props to specify default values using a factory function, which would alleviate the need for this assignment and `internalProxyManager`.

File: src/app.js [hunk 3]
```diff
@@ -12,13 +12,9 @@ import 'material-design-icons-iconfont/dist/material-design-icons.css';
 
 import 'paraview-glance/src/io/ParaViewGlanceReaders';
 import ReaderFactory from 'paraview-glance/src/io/ReaderFactory';
-import App from 'paraview-glance/src/components/core/App';
+import App, { $globalBus } from 'paraview-glance/src/components/core/App';
```
Comment (zachmullen) [inline src/app.js]: Might be worth changing the name to `$appBus` or something? I don't have strong feelings, though.

File: src/app.js (hunk 1 above)
Comment (floryst) [reply src/app.js]: Good catch, will remove that.

**Comment by floryst** [comment]
I've been looking at removing `$globalBus` and solely using vuex. Right now the global bus is used for actions against other components rather than transmitting state, so I need some more time to work out how to drop the event bus altogether.

**Comment by floryst** [comment]
This turned out to be a large change-set. Anyways, I've removed `$globalBus` in favor of a vuex store, so now state is more centralized. @zachmullen I don't know if you have a better or more preferred way of handling vuex mutation/action/getter names (see `stores/types.js`), so suggestions would be greatly appreciated.

**Comment by zachmullen** [comment]
I don't have much preference on that. One thing worth considering (some believe it's a best practice) is to always use actions rather than mapping mutations directly into components; this allows a uniform Promise-based interface into the store, and allows actions to internally switch between synchronous / asynchronous without becoming a breaking change. The downside is more boilerplate.

I don't have strong feelings one way or the other, and it doesn't need to be done on this branch either way, just worth thinking about.

**Comment by zachmullen** [comment]
I'm fine to ship this as-is and iron out the kinks on subsequent (smaller) branches.

**Comment by floryst** [comment]
Sounds good. I've also re-added history based routing, but now I'm using the history API (I was hoping to scrape by with hashes, but nope), and that logic is in app.js and not in the <App> component.

**Comment by zachmullen** [comment]
Great, thanks!

**Pull request merged**